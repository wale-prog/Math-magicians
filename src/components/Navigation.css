.nav-list {
  display: flex;
  gap: 10px;
}

body {
  background-color: #d3d3d3;
  color: rgb(72, 33, 33);
}

.nav-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: rgb(241, 223, 217);
  margin-bottom: 48px;
  padding: 0 6rem 0 3rem;
}

.nav-item {
  padding-right: 8px;
  list-style: none;
  border-right: 2px solid;
}

.nav-item a {
  text-decoration: none;
  color: #000;
  transition: all 0.3s ease-in-out;
}

.nav-item a:hover {
  color: rgb(240, 143, 143);
  font-size: 18px;
}

.nav-item:last-child {
  border-right: none;
}

.heading-text {
  text-decoration: none;
  font-size: 32px;
  font-weight: bold;
  color: rgb(72, 33, 33);
  cursor: pointer;
  padding: 18px 0 18px 0;
}
