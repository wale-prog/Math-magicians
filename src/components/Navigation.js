/* eslint-disable */

import React from "react";
import { NavLink } from "react-router-dom";
import "./Navigation.css";

const Navigation = () => {
  return (
    <div className="nav-container">
      <h1>Math Magicians</h1>
      <ul className="nav-list">
        <li className="nav-item">
          <NavLink activeStyle={{ color: "red" }} to="/welcome">
            Home
          </NavLink>
        </li>
        <li className="nav-item">
          <NavLink activeStyle={{ color: "red" }} to="/calculator">
            Calculator
          </NavLink>
        </li>
        <li className="nav-item">
          <NavLink activeStyle={{ color: "red" }} to="/quotes">
            Quote
          </NavLink>
        </li>
      </ul>
    </div>
  );
};
export default Navigation;
