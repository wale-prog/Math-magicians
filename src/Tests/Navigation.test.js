import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

describe('Testing the Navigation', () => {
  it('Navigation is rendered', () => {
    const nav = renderer
      .create(<BrowserRouter><Navigation /></BrowserRouter>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
