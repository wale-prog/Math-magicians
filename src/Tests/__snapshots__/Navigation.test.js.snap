// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Navigation Navigation is rendered 1`] = `
<div
  className="nav-container"
>
  <a
    aria-current={null}
    className="heading-text"
    href="/welcome"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="nav-list"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        href="/welcome"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        href="/quotes"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</div>
`;
